lib/chef/knife/ssh.rb

```diff
@@ -160,6 +160,31 @@ def configure_session
         session_from_list(list)
       end
 
+      def get_ssh_attribute(node)
+        # Order of precedence for ssh target
+        # 1) command line attribute
+        # 2) configuration file
+        # 3) cloud attribute
+        # 4) fqdn
+        if config[:attribute_from_cli]
```

Comment:
Looks like that was the only change needed.
I have done some extra testing of the latest code with various combinations of cli and config. 
The priority behaviour works as expected.
1 commandline
2 config file
3 cloud

Thanks!